### PR TITLE
Rename gmaven_rules to rules_jvm_external

### PR DESCRIPTION
**Pull request by jin**
TODO: Rename this repository in lockstep while merging this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,12 +24,12 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 GMAVEN_RULES_TAG = "0.1.0" # or latest tag
 
 http_archive(
-    name = "gmaven_rules",
-    strip_prefix = "gmaven_rules-%s" % GMAVEN_RULES_TAG,
-    url = "https://github.com/bazelbuild/gmaven_rules/archive/%s.zip" % GMAVEN_RULES_TAG,
+    name = "rules_jvm_external",
+    strip_prefix = "rules_jvm_external-%s" % GMAVEN_RULES_TAG,
```
Comment (dkelmer) [inline README.md]: switch GMAVEN_RULES_TAG too

**Comment by jin** [comment]
The users of legacy gmaven_rules are using/pinning https://github.com/bazelbuild/gmaven_rules/releases/tag/20181212-2

After this, we'll rename the repository to rules_jvm_external, and create a new release containing the new and legacy gmaven_rules. Then we can update the Android projects to use that release.

**Comment by meteorcloudy** [comment]
@jin Looks like this one broke https://buildkite.com/bazel/android-testing/builds/231#b88fef83-4039-4bce-9514-ca5630f89e64

**Comment by jin** [comment]
Yep, fix is here: 
https://github.com/googlesamples/android-testing/pull/249

**Comment by meteorcloudy** [comment]
https://buildkite.com/bazel/rules-foreign-cc/builds/273#2f3c5420-5cfe-4415-9e4e-47607aa6279a
Can you also help rules-foreign-cc migrate?

**Comment by jin** [comment]
@meteorcloudy 
done: https://github.com/bazelbuild/rules_foreign_cc/pull/236

**Pull request merged**